blender/arm/material/make_mesh.py

```diff
@@ -536,15 +542,28 @@ def make_forward(con_mesh):
                 frag.add_uniform('sampler2DShadow shadowMapSpot[4]', included=True)
 
     if not blend:
-        mrt = rpdat.rp_ssr  # mrt: multiple render targets
-        if mrt:
+        mrt = 0  # mrt: multiple render targets
+        if rpdat.rp_ssr:
+            mrt += 1
+        if rpdat.rp_ss_refraction:
+            mrt += 1
+        if mrt != 0:
+            idx_refraction = 1
             # Store light gbuffer for post-processing
-            frag.add_out('vec4 fragColor[2]')
+            frag.add_out(f'vec4 fragColor[{mrt}+1]')
             frag.add_include('std/gbuffer.glsl')
             frag.write('n /= (abs(n.x) + abs(n.y) + abs(n.z));')
             frag.write('n.xy = n.z >= 0.0 ? n.xy : octahedronWrap(n.xy);')
             frag.write('fragColor[0] = vec4(direct + indirect, packFloat2(occlusion, specular));')
-            frag.write('fragColor[1] = vec4(n.xy, roughness, metallic);')
+            if rpdat.rp_ssr:
+                frag.write('fragColor[1] = vec4(n.xy, roughness, metallic);')
```

Comment:
Are you sure that this line should only be executed if `rpdat.rp_ssr` is true?

Reply:
Yes true, refraction uses the roughness too.